backend/src/strategies/strategies.service.ts

```diff
@@ -0,0 +1,105 @@
+import { HttpException, HttpStatus, Injectable, Logger, NestMiddleware } from "@nestjs/common";
+import { Request, Response } from "express";
+import { ImsUserFindingService } from "src/backend-services/ims-user-finding.service";
+import { StrategyInstance } from "src/model/postgres/StrategyInstance.entity";
+import { StrategyInstanceService } from "src/model/services/strategy-instance.service";
+import { PerformAuthFunctionService } from "./perform-auth-function.service";
+import { StrategiesService as StrategiesRepository } from "../model/services/strategies.service";
+import { Strategy } from "./Strategy";
+import { OAuthHttpException } from "src/errors/OAuthHttpException";
+import { AuthException } from "src/errors/AuthException";
+import { Context, FlowState } from "../util/Context";
+import { ActiveLogin } from "../model/postgres/ActiveLogin.entity";
+import { compareTimeSafe } from "../util/utils";
+
+@Injectable()
+export class StrategiesService implements NestMiddleware {
```

Comment:
is the implements NestMiddleware necessary? a service implementing Middleware seems a bit weird to me